Item(by='brundolf', descendants=None, kids=[24731595], score=None, time=1602260994, title=None, item_type='comment', url=None, parent=24730194, text='This is an interesting case for error-handling, because a) it can happen virtually <i>anywhere</i>, so handling it with Result would add huge API overhead in terms of ergonomics (given that most code never really has to think about it), yet b) it&#x27;s very important that it <i>is</i> able to be handled in certain kinds of system contexts, which panics are not designed to facilitate. Most error cases seem to fall neatly into one camp or the other (something you always want to explicitly handle, or something where you just want to abort), but this one doesn&#x27;t.<p>handle_alloc_error seems to do well enough as a workaround, but (from my superficial reading of the GitHub thread) it feels like just a very specific &quot;poor-man&#x27;s try&#x2F;catch&quot; for this one particular case. It <i>feels</i> like a workaround.<p>In general I&#x27;m a big fan of Result&#x2F;panic in place of traditional exceptions, but this usecase makes it really quite unideal')